File: 1402-reducing-dishes/1402-reducing-dishes.cpp
class Solution {
public:
    int maxSatisfaction(vector<int>& satisfaction) {
        sort(satisfaction.begin(), satisfaction.end());
        int ans = 0;
        
        for (int i=0; i < satisfaction.size(); i++){
            int currSum = satisfaction[i];
            int coff = 2;
            for (int j=i+1; j < satisfaction.size(); j++){
                currSum += (satisfaction[j] * coff);
                coff += 1;
                    
            }
            ans = max(ans, currSum);
        }
        return ans;
    }
};

File: 1402-reducing-dishes/1402-reducing-dishes.py
class Solution:
    def maxSatisfaction(self, nums: List[int]) -> int:
       
        nums.sort()
        ans = 0 
        for i, num in enumerate(nums):
            currsum = num
            coff = 2
            for j in range(i+1, len(nums)):
                currsum += (coff * nums[j])
                coff += 1
            ans = max(ans, currsum)
        return ans                